be/src/exec/vectorized/arrow_to_json_converter.cpp

```diff
@@ -0,0 +1,337 @@
+// This file is made available under Elastic License 2.0.
+
+#include <memory>
+#include <string>
+
+#include "arrow/array.h"
+#include "arrow/type.h"
+#include "arrow/type_fwd.h"
+#include "arrow/type_traits.h"
+#include "column/json_column.h"
+#include "gutil/casts.h"
+#include "gutil/strings/substitute.h"
+#include "util/json.h"
+
+namespace starrocks::vectorized {
+
+using namespace arrow;
+using ArrowStatus = Status;
+
+// Convert multi-row Array to JsonColumn
+static Status convert_multi_arrow_list(const ListArray* array, JsonColumn* output);
+static Status convert_multi_arrow_struct(const StructArray* array, JsonColumn* output);
+static Status convert_multi_arrow_map(const MapArray* array, JsonColumn* output);
+static Status convert_multi_arrow_primitive(const Array* array, JsonColumn* output);
+
+// Convert a single row in Array to json
+static Status convert_single_arrow_list(const ListArray* array, int offset, vpack::Builder* builder);
+static Status convert_single_arrow_struct(const StructArray* array, int offset, vpack::Builder* builder);
+static Status convert_single_arrow_map(const MapArray* array, int offset, vpack::Builder* builder);
+
+// Convert the whole array to json array
+static Status convert_arrow_to_json_array(const Array* array, Type::type value_type, vpack::Builder* output);
+static Status convert_arrow_to_json_array(const StringArray* array, vpack::Builder* builder);
+static Status convert_arrow_to_json_array(const BooleanArray* array, vpack::Builder* builder);
+
+// Convert the element at offset to a json element, which could be both primitive or nested
+static Status convert_arrow_to_json_element(const Array* array, Type::type type_id, int offset, const std::string& name,
+                                            vpack::Builder* output);
+
+template <class TypeClass, class CType = typename TypeClass::c_type>
+static enable_if_number<TypeClass, Status> convert_arrow_to_json_array(const NumericArray<TypeClass>* array,
+                                                                       vpack::Builder* builder) {
+    builder->openArray();
+    for (int i = 0; i < array->length(); i++) {
+        CType value = array->Value(i);
+        builder->add(vpack::Value(value));
```

Comment:
That's a good suggestion.

But currently vpack has no interface to build an primitive array one-time, so I think it could be leaved to future optimization.